test/test_utils.js

```diff
@@ -321,6 +321,10 @@ describe("Testing utils functions", function() {
             // 'Invalid date' when input length < 4
             assert.deepEqual(utils.extract_za_id_dob("810"), "Invalid date");
         });
+        it("", function() {
+            assert.deepEqual(utils.extract_za_id_dob("5202017805280"),
+                moment("1952-02-01").format("YYYY-MM-DD"));
+        });
```

Comment:
Please add more tests, especially at the boundaries - like last day of 1948, first day of 1949, where the boundary used to be (around 1968), and year 2000.